packages/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -88,7 +94,9 @@ export const useGridKeyboardNavigation = (
           colIndex = nextCellColSpanInfo.rightVisibleCellIndex;
         }
       }
-      const field = gridVisibleColumnFieldsSelector(apiRef)[colIndex];
+      const field = listView
+        ? gridVisibleListColumnFieldsSelector(apiRef)[colIndex]
+        : gridVisibleColumnFieldsSelector(apiRef)[colIndex];
```

Comment:
Do you think we should encapsulate this logic in a single selector? We had already talked about giving selectors access to some props, this could be a good candidate, if it's logic that will need to be repeated a lot.

Reply:
Added to https://github.com/mui/mui-x/issues/14997.